From 370988db68e33e8414717e95e816b9c2ec2635a5 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Thu, 2 Aug 2018 13:43:08 -0700
Subject: [PATCH] TRITON-621 mockcloud delegate dataset support (dummy backend
 will use 'mockcloudRoot' mdata for mock server data, if available)

---
 lib/backends/dummy/common.js              | 14 +++++++++++++-
 lib/backends/dummy/tools/create-server.js |  2 +-
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/lib/backends/dummy/common.js b/lib/backends/dummy/common.js
index 5c8efaa..e64c67c 100644
--- a/lib/backends/dummy/common.js
+++ b/lib/backends/dummy/common.js
@@ -21,7 +21,19 @@ var fs = require('fs');
 var assert = require('assert-plus');
 var async = require('async');
 
-var SERVER_ROOT = '/opt/custom/virtual/servers';
+var mockcloudRoot;
+try {
+    mockcloudRoot = child_process
+        .execSync('/usr/sbin/mdata-get mockcloudRoot', {encoding: 'utf8'})
+        .trim();
+} catch (err) {
+    // The old default for backward compatibility.
+    mockcloudRoot = '/opt/custom/virtual';
+    console.warn('warning: cn-agent dummy backend could not get ' +
+        '"mockcloudRoot" dir from mdata, using default %s: %s',
+        mockcloudRoot, err);
+}
+var SERVER_ROOT = mockcloudRoot + '/servers';
 
 function mdataGet(key, callback) {
     assert.string(key, 'key');
diff --git a/lib/backends/dummy/tools/create-server.js b/lib/backends/dummy/tools/create-server.js
index e680291..e8e8ec1 100644
--- a/lib/backends/dummy/tools/create-server.js
+++ b/lib/backends/dummy/tools/create-server.js
@@ -286,7 +286,7 @@ function writeSysinfo(ctx, callback) {
 
     fs.writeFile(filename, data, function _wroteSysinfo(err) {
         if (!err) {
-            console.log('wrote sysinfo');
+            console.log('wrote sysinfo (%s)', filename);
         }
 
         callback(err);
-- 
2.21.0

